### PR TITLE
fix: LSDV-5071: Ensure secret key is securely set persisting a new one if required

### PR DESCRIPTION
**Pull request by bmartel**
### PR fulfills these requirements
- [X] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend
- [X] CLI

### CODE REVIEW

File: label_studio/core/utils/io.py [hunk 1]
```diff
@@ -208,3 +210,20 @@ def validate_upload_url(url, block_local_urls=True):
...
+
+    if value == "":
+        print(f'Warning: {key} not found in environment variables will generate a random key')
+        value = get_random_secret_key()
+        try:
+            env_file = os.path.join(base_dir, '..', '..', '.env')
+            with open(env_file, 'a') as f:
+                f.write(f'\n{key}={value}')
+        except Exception as e:
+            print(f'Warning: failed to write {key} to .env file: {e}, new key will be regenerated on every run')
```
Comment (wesleylima) [inline label_studio/core/utils/io.py]: Maybe add more context on the impact about it being generated every run? Sessions will be reset at every restart, encrypted values (do we have that in LS?) will be undecryptable... Does this affect password hashes?

File: label_studio/core/utils/io.py [hunk 2]
```diff
@@ -217,7 +218,6 @@ def generate_key_if_missing(key):
         print(f'Warning: {key} not found in environment variables will generate a random key')
         value = get_random_secret_key()
         try:
-            env_file = os.path.join(base_dir, '..', '..', '.env')
             with open(env_file, 'a') as f:
                 f.write(f'\n{key}={value}')
```
Comment (jombooth) [inline label_studio/core/utils/io.py]: I would suggest adding a trailing newline after the value as well

**Review by jombooth — APPROVED** [review]

File: label_studio/core/utils/io.py (hunk 1 above)
Comment (jombooth) [reply label_studio/core/utils/io.py]: https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-SECRET_KEY <- "Secret keys are not used for passwords of users and key rotation will not affect them."

adding more context here SGTM also

**Review by dredivaris — APPROVED** [review]
LGTM

**Pull request merged**